src/Mono.Android/Xamarin.Android.Net/AndroidClientHandler.cs

```diff
@@ -285,7 +287,7 @@ await stream.CopyToAsync(httpConnection.OutputStream, 4096, cancellationToken)
 			}
 
 			if (cancellationToken.IsCancellationRequested) {
-				httpConnection.Disconnect();
+				await Task.Run (() => httpConnection.Disconnect()).ConfigureAwait (false);
```

Comment:
Given that we do this in no less than three places, maybe we should have a helper?

    async Task Disconnect (HttpURLConnection httpConnection)
    {
        await Task.Run (() => httpConnection.Disconnect())
            .ConfigureAwait (continueOnCapturedContext: false);
    }
    ...
    await Disconnect (httpConnection);